### PR TITLE
Support negative number in config

### PR DESCRIPTION
**Pull request by Kahbazi**
Fixes https://github.com/microsoft/reverse-proxy/issues/565

### CODE REVIEW

File: src/ReverseProxy/Utilities/ConfigurationReadingExtensions.cs [hunk 1]
```diff
@@ -9,7 +9,7 @@ internal static class ConfigurationReadingExtensions
     {
         internal static int? ReadInt32(this IConfiguration configuration, string name)
         {
-            return configuration[name] is string value ? int.Parse(value, NumberStyles.None, CultureInfo.InvariantCulture) : null;
+            return configuration[name] is string value ? int.Parse(value, NumberStyles.AllowLeadingSign, CultureInfo.InvariantCulture) : null;
```
Comment (Kahbazi) [inline src/ReverseProxy/Utilities/ConfigurationReadingExtensions.cs:12]: Is `NumberStyles.AllowLeadingSign` good enough, or Should it be `NumberStyles.Integer` which has `AllowLeadingWhite`, `AllowTrailingWhite`, and `AllowLeadingSign`?

**Review by Tratcher — COMMENTED** [review]
Nice. Can you also update the test for Order to use a negative value?
https://github.com/microsoft/reverse-proxy/blob/b34e66a4394e6c313f6f88331d049edc6fadadd3/test/ReverseProxy.Tests/Configuration/ConfigurationConfigProviderTests.cs#L307

**Review by Tratcher — APPROVED** [review]

**Comment by Tratcher** [comment]
Thanks
